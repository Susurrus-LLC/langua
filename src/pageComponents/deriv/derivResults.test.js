import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import DerivResults from './derivResults'

describe('DerivResults', () => {
  const styles = {
    outputBox: 'outputBox',
    term: 'term',
    affixGloss: 'affixGloss',
    gloss: 'gloss',
    hoveredGloss: 'hoveredGloss',
    lexGloss: 'lexGloss',
    derivRow: 'derivRow',
    outputRow: 'outputRow',
    rowNumber: 'rowNumber',
    affixDef: 'affixDef',
    output: 'output'
  }

  const results = {
    newWords: [
      {
        lexeme: {
          lexeme: 'nzi',
          definition: 'tree'
        },
        suffix: {
          derivation: '-vo',
          gloss: 'person',
          definition:
            'Creates a noun referring to a person associated with the base.'
        }
      },
      {
        lexeme: {
          lexeme: 'lÉ›tsi',
          definition: 'sword'
        },
        prefix: {
          derivation: 'k-',
          gloss: 'collective',
          definition: 'Creates a noun referring to a collection of bases.'
        },
        suffix: {
          derivation: '-sa',
          gloss: 'collective',
          definition: 'Creates a noun referring to a collection of bases.'
        }
      }
    ],
    possible: 25
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<DerivResults styles={styles} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately with various props', () => {
    let tree = renderer.create(<DerivResults styles={styles} />).toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderer
      .create(<DerivResults styles={styles} results={results} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
