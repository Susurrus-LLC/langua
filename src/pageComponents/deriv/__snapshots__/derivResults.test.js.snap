// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DerivResults renders accurately with various props 1`] = `
<div
  className="results"
>
  <div
    className="output"
  />
  <div
    className="stats"
  >
    <p
      className="statsText"
    >
      words: 0; maximum derivations possible: 0
    </p>
  </div>
</div>
`;

exports[`DerivResults renders accurately with various props 2`] = `
<div
  className="results"
>
  <div
    className="output"
  >
    <div
      className="derivRow outputRow"
    >
      <div
        className="outputBox rowNumber"
      >
        1
        .
      </div>
      <div>
        <div
          className="derivRow outputRow"
        >
          <div
            className="outputBox"
          >
            <p
              className="term"
            >
              nzi
            </p>
            <p
              className="lexGloss gloss"
            >
              tree
            </p>
          </div>
          <div
            className="outputBox"
          >
            <p
              className="term"
            >
              -vo
            </p>
            <p
              className="affixGloss gloss"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              person
            </p>
          </div>
        </div>
        <p
          className="affixDef"
        >
          Creates a noun referring to a person associated with the base.
        </p>
      </div>
    </div>
    <div
      className="derivRow outputRow"
    >
      <div
        className="outputBox rowNumber"
      >
        2
        .
      </div>
      <div>
        <div
          className="derivRow outputRow"
        >
          <div
            className="outputBox"
          >
            <p
              className="term"
            >
              k-
            </p>
            <p
              className="affixGloss gloss"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              collective
            </p>
          </div>
          <div
            className="outputBox"
          >
            <p
              className="term"
            >
              l…õtsi
            </p>
            <p
              className="lexGloss gloss"
            >
              sword
            </p>
          </div>
          <div
            className="outputBox"
          >
            <p
              className="term"
            >
              -sa
            </p>
            <p
              className="affixGloss gloss"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              collective
            </p>
          </div>
        </div>
        <p
          className="affixDef"
        >
          Creates a noun referring to a collection of bases.
        </p>
      </div>
    </div>
  </div>
  <div
    className="stats"
  >
    <p
      className="statsText"
    >
      words: 2; maximum derivations possible: 25
    </p>
  </div>
</div>
`;
